src/plugins/ui_actions/public/actions/action_internal.ts

```diff
@@ -66,7 +66,13 @@ export class ActionInternal<Context extends object = object>
 
   public async isCompatible(context: Context): Promise<boolean> {
     if (!this.definition.isCompatible) return true;
-    return await this.definition.isCompatible(context);
+    try {
+      return await this.definition.isCompatible(context);
+    } catch (e) {
+      // eslint-disable-next-line no-console
+      console.error(e);
```

Comment:
How about logging error once? This error could happen numerous times and we want to avoid flooding console. isCompatible is called a lot